pkg/globalnet/controllers/controllers_suite_test.go

```diff
@@ -142,6 +142,24 @@ func (t *testDriverBase) afterEach() {
 	t.controller.Stop()
 }
 
+func (t *testDriverBase) initChains() {
```

Comment:
Agree that with this change tests code is aligned with real nftables/iptables behavior.

the new initChains function should be synced with [GN chains creation function](https://github.com/submariner-io/submariner/blob/d405b10dad8dc7b973435ef3ce70163bc6a78f9f/pkg/globalnet/controllers/gateway_monitor.go#L454),  if we can't run 'real' GN chains creation function here, maybe we should at least add a comment saying that this function should be synced with real function

Reply:
I'm not sure what you mean by synced with but I think you're concern is re: not running the real function. That is in the `gatewayMonitor` which is tested in _gateway_monitor_test.go_. In those tests we do not call `initChains` and also verify that the chains are created by the `gatewayMonitor`.

Reply:
Yep, was concerned that future changes to gateway_monitor createGlobalnetChains  or fake packetfilter implementation (e.g: different implementation for CreateIPHookChainIfNotExists and CreateChainIfNotExists) will break testing.

On the other hand, since we don't expect changes in GN chains, maybe we can leave it as you suggested

Reply:
yeah if `createGlobalnetChains` is changed, eg add a new chain, then the unit tests would need to be adjusted accordingly but that should be expected.